src/mapgen_v7.cpp

```diff
@@ -450,7 +429,7 @@ int MapgenV7::generateTerrain()
 			if (vm->m_data[vi].getContent() == CONTENT_IGNORE) {
 				if (y <= surface_y) {
 					vm->m_data[vi] = n_stone;  // Base terrain
-				} else if (mountain_flag &&
+				} else if ((spflags & MGV7_MOUNTAINS) &&
```

Comment:
'bool mountain_flag' should be used here instead of running a bitwise operation 500,000 times per chunk.